main.qml

```diff
@@ -794,4 +794,8 @@ ApplicationWindow {
             }
         }
     }
+    onClosing: {
+        walletManager.closeWallet(currentWallet);
+        console.log("onClosing called");
+    }
```

Comment:
@Jaqueeee that must be a bug - missing "cleanup" code.  Wallet lifetime managed only by WalletManager, so only the way to properly destruct the wallet object is WalletManager::closeWallet() call. Probably make sense to implement some ref counting within WalletManager and close wallet (wallets) by e.g. QCoreApplication::aboutToQuit() signal